src/utils/components/ColumnManagementModal/ColumnManagementModal.tsx

```diff
@@ -0,0 +1,192 @@
+import React, { FC, MouseEventHandler } from 'react';
+
+import { useKubevirtTranslation } from '@kubevirt-utils/hooks/useKubevirtTranslation';
+import useKubevirtUserSettingsTableColumns from '@kubevirt-utils/hooks/useKubevirtUserSettings/useKubevirtUserSettingsTableColumns';
+import { ColumnLayout } from '@openshift-console/dynamic-plugin-sdk';
+import {
+  ActionList,
+  ActionListItem,
+  Alert,
+  AlertVariant,
+  Button,
+  ButtonVariant,
+  DataList,
+  Form,
+  Modal,
+  ModalVariant,
+  Stack,
+  StackItem,
+} from '@patternfly/react-core';
+
+import { MAX_VIEW_COLS } from './constants';
+import DataListRow from './DataListRow';
+
+import './column-management-modal.scss';
+
+type ColumnManagementModalProps = {
+  columnLayout: ColumnLayout;
+  isOpen: boolean;
+  onClose: () => void;
+};
+
+export const ColumnManagementModal: FC<ColumnManagementModalProps> = ({
+  columnLayout,
+  isOpen,
+  onClose,
+}) => {
+  const { t } = useKubevirtTranslation();
+  const defaultColumns = columnLayout.columns.filter((column) => column.id && !column.additional);
+  const additionalColumns = columnLayout.columns.filter((column) => column.additional);
+
+  const [_, setActiveColumns, loaded, error] = useKubevirtUserSettingsTableColumns({
+    columnManagementID: columnLayout.id,
+    columns: columnLayout.columns,
+  });
+
+  const [checkedColumns, setCheckedColumns] = React.useState<Set<string>>(
+    columnLayout.selectedColumns && columnLayout.selectedColumns.size !== 0
+      ? new Set(columnLayout.selectedColumns)
+      : new Set(defaultColumns.map((col) => col.id)),
+  );
+
+  const onColumnChange = (checked: boolean, event: React.SyntheticEvent): void => {
+    const updatedCheckedColumns = new Set<string>(checkedColumns);
+    const selectedId = event?.currentTarget?.id?.slice(4);
+    updatedCheckedColumns.has(selectedId)
+      ? updatedCheckedColumns.delete(selectedId)
+      : updatedCheckedColumns.add(selectedId);
+    setCheckedColumns(updatedCheckedColumns);
+  };
+
+  const submit: MouseEventHandler<HTMLButtonElement> = async (event) => {
+    event.preventDefault();
+    const orderedCheckedColumns = new Set<string>();
+    columnLayout.columns.forEach(
+      (column) => checkedColumns.has(column.id) && orderedCheckedColumns.add(column.id),
+    );
```

Comment:
Can you please check if checkedColumes.forEach((col) => orderedCheckedColumns.add(col.id)) works the same?